tests/TestCase/ORM/QueryRegressionTest.php

```diff
@@ -487,6 +488,35 @@ public function testDeepBelongsToManySubqueryStrategy()
     }
 
     /**
+     * Tests that using the subquery strategy in a deep association returns the right results
+     *
+     * @see https://github.com/cakephp/cakephp/issues/5769
+     * @return void
+     */
+    public function testDeepBelongsToManySubqueryStrategy2()
+    {
+        $table = TableRegistry::get('Authors');
+        $table->hasMany('Articles');
+        $table->Articles->belongsToMany('Tags', [
+            'strategy' => 'subquery'
+        ]);
+        $table->belongsToMany('Tags', [
+            'strategy' => 'subquery',
+        ]);
+        $table->Articles->belongsTo('Authors');
+
+        $result = $table->Articles->find()
```

Comment:
No, the error could only be triggered when querying from the HasMany association.